src/js/components/FormField/FormField.js

```diff
@@ -451,7 +459,22 @@ const FormField = forwardRef(
         {...outerProps}
         style={outerStyle}
         onFocus={(event) => {
-          setFocus(shouldKeepFocus());
+          setFocus(containsFocus(formFieldRef.current));
+          const formDescendants =
+            formFieldRef.current.getElementsByTagName('*');
+          let focusableDescendants = 0;
+          for (let i = 0; i < formDescendants.length; i += 1) {
+            const child = formDescendants[i];
+            if (isFocusable(child)) {
```

Comment:
I like this generic. It seems more generic and more likely to work with random components.

One question I have is how does this work with Drops which may not be a descendent? Is this an issue?

Reply:
When focus transfers to content in the Drop this is handled by `setFocus(containsFocus(formFieldRef.current));` (line 462 in FormField.js) so it is not an issue that the Drop isn't included in formDescendants. (Note: I tested with Select and DateInput, both are built using Drop and focus correctly transfers.)

Reply:
Great. Thanks.